### PR TITLE
Auto-generate: Optimize panel title / description generation

### PR DESCRIPTION
**Pull request by SandersAaronD**
**What is this feature?**

Reduces token count and improves results by filtering noisy keys from requests to OpenAI for panel title and description.

**Why do we need this feature?**

Reduce costs and improve panel title/description generation.

**Who is this feature for?**

Anyone using DashGPT features.

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: public/app/features/dashboard/components/GenAI/utils.ts [hunk 1]
```diff
@@ -121,3 +119,29 @@ export function getDashboardPanelPrompt(dashboard: DashboardModel): string {
   // So it is possibly that if we can condense it further it would be better
   return panelPrompt;
 }
+
+export function getFilteredPanelString(panel: PanelModel): string {
```
Comment (nmarrs) [inline public/app/features/dashboard/components/GenAI/utils.ts:123]: we could probably incorporate this in the dashboard title / description generation as well as that also takes in panel info (currently limited to first 10 panels with descriptions) - can be a follow-up though

**Review by nmarrs — APPROVED** [review]
🧪

File: public/app/features/dashboard/components/GenAI/utils.ts [hunk 2]
```diff
@@ -121,3 +119,29 @@ export function getDashboardPanelPrompt(dashboard: DashboardModel): string {
...
+    'description',
+    'targets',
+    'thresholds',
+    'type',
+    'xaxis',
+    'yaxes',
+  ]);
+
+  // This cannot avoid the use of any because the type of panelObj is any
+  const panelObjFiltered = Object.keys(panelObj).reduce((obj: { [key: string]: any }, key) => {
```
[the remark this answers is not in the document]
Comment (SandersAaronD) [reply public/app/features/dashboard/components/GenAI/utils.ts:139]: I'll put that correction into my next PR.

File: public/app/features/dashboard/components/GenAI/utils.ts (hunk 1 above)
Comment (SandersAaronD) [reply public/app/features/dashboard/components/GenAI/utils.ts:123]: Dashboard generation I think presently only gets titles and descriptions, not anything else. Only panel was sending the entire panel json.

**Pull request merged**